tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_overwrite_oldest.c

```diff
@@ -0,0 +1,241 @@
+	bt_conn_foreach_fake.custom_fake = bt_conn_foreach_custom_fake_key_slot_0_in_use;
+
+	/* 2 calls are required as the first slot will be considered in use
+	 * with a connection by the effect of 'bt_conn_foreach_custom_fake()'
+	 */
+	update_oldest_params_vector();
+	oldest_params_ref_idx = update_oldest_params_vector();
+
+	returned_key = bt_keys_get_addr(id, &addr);
+
+	zassert_true(returned_key != NULL, "bt_keys_get_addr() returned a NULL reference");
+	zassert_true(returned_key == returned_keys_refs[oldest_params_ref_idx],
+		     "bt_keys_get_addr() returned reference doesn't match expected one");
+}
+
+/*
+ *  Test adding extra (ID, Address) pair while the keys pool list
+ *  is full
+ *
+ *  Constraints:
+ *   - Keys pool list is full
+ *   - Key at slot 0 is used with a connection
+ *   - Oldest key isn't used with a connection
+ *   - CONFIG_BT_KEYS_OVERWRITE_OLDEST is enabled
+ *
+ *  Expected behaviour:
+ *   - Valid reference pointer is returned which matches the oldest
+ *     reference with index 2
+ */
+void test_add_item_full_list_key_0_in_use_key_2_oldest(void)
+{
+	bt_addr_le_t addr;
+	struct bt_keys *returned_key;
+	uint8_t id = BT_ADDR_ID_4;
+	uint32_t oldest_params_ref_idx;
+
+	bt_addr_le_copy(&addr, (const bt_addr_le_t *)BT_ADDR_LE_4);
+
+	oldest_params_ref_idx = update_oldest_params_vector();
+	bt_unpair_fake.custom_fake = bt_unpair_custom_fake;
+	bt_conn_foreach_fake.custom_fake = bt_conn_foreach_custom_fake_key_slot_0_in_use;
+
+	returned_key = bt_keys_get_addr(id, &addr);
+
+	zassert_true(returned_key != NULL, "bt_keys_get_addr() returned a NULL reference");
+	zassert_true(returned_key == returned_keys_refs[oldest_params_ref_idx],
+		     "bt_keys_get_addr() returned reference doesn't match expected one");
+}
+
+/*
+ *  Test adding extra (ID, Address) pair while the keys pool list
+ *  is full
+ *
+ *  Constraints:
+ *   - Keys pool list is full
+ *   - All Keys are used with a connection
+ *   - CONFIG_BT_KEYS_OVERWRITE_OLDEST is enabled
+ *
+ *  Expected behaviour:
+ *   - Valid reference pointer is returned which matches the oldest
+ *     reference with index 2
+ */
+void test_add_item_full_list_all_keys_in_use(void)
+{
+	bt_addr_le_t addr;
+	struct bt_keys *returned_key;
+	uint8_t id = BT_ADDR_ID_5;
+
+	bt_addr_le_copy(&addr, (const bt_addr_le_t *)BT_ADDR_LE_5);
+
+	bt_unpair_fake.custom_fake = bt_unpair_unreachable_custom_fake;
+	bt_conn_foreach_fake.custom_fake = bt_conn_foreach_custom_fake_all_keys_in_use;
+
+	returned_key = bt_keys_get_addr(id, &addr);
+
+	zassert_true(returned_key == NULL, "bt_keys_get_addr() returned a non-NULL reference");
+}
+
+/* Setup test variables */
+static void unit_test_setup(void)
+{
+	/* Register resets */
+	CONN_FFF_FAKES_LIST(RESET_FAKE);
+	HCI_CORE_FFF_FAKES_LIST(RESET_FAKE);
+}
+
+static bool pragma_overwrite_enabled(const void *s)
+{
+	if ((IS_ENABLED(CONFIG_BT_KEYS_OVERWRITE_OLDEST))) {
+		return true;
+	}
+
+	return false;
+}
+
+/* Tests need to be kept in the same order */
```

Comment:
I really don't like this comment. Refer to the one in `keys_get_addr/src/main.c`.

Reply:
This test suite verifies the order at which the oldest key is overwritten.
So, it's actually required to run test cases in the correct order so that we can expect the output.